src/libraries/System.Net.Http.WinHttpHandler/tests/FunctionalTests/WinHttpHandlerTest.cs

```diff
@@ -172,10 +173,97 @@ public async Task GetAsync_SetCookieContainerMultipleCookies_CookiesSent()
             };
         }
 
+        [OuterLoop("Uses delays")]
+        [Fact]
+        public async Task SendAsync_MultipleHttp2ConnectionsEnabled_CreateAdditionalConnections()
+        {
+            // Warm up thread pool because the full .NET Framework calls synchronous Stream.Read() and we need to delay those calls thus threads will get blocked.
+            ThreadPool.GetMinThreads(out _, out int completionPortThreads);
+            ThreadPool.SetMinThreads(401, completionPortThreads);
+            using (WinHttpHandler handler = new WinHttpHandler())
+            {
+                handler.EnableMultipleHttp2Connections = true;
+                handler.ServerCertificateValidationCallback = HttpClientHandler.DangerousAcceptAnyServerCertificateValidator;
+                const int maxActiveStreamsLimit = 100 * 3;
+                const string payloadText = "Multiple HTTP/2 connections test.";
+                TaskCompletionSource<bool> delaySource = new TaskCompletionSource<bool>();
+                using (HttpClient client = new HttpClient(handler))
+                {
+                    List<(Task<HttpResponseMessage> task, DelayedStream stream)> requests = new List<(Task<HttpResponseMessage> task, DelayedStream stream)>();
+                    for (int i = 0; i < maxActiveStreamsLimit; i++)
+                    {
+                        HttpRequestMessage request = new HttpRequestMessage(HttpMethod.Post, Configuration.Http.Http2RemoteEchoServer) { Version = HttpVersion20.Value };
```

Comment:
Why use a remote server here? I think we should be using loopback whenever possible.

Loopback would also improve this test by being able to assert that the max stream limit is hit and a 2nd connection is opened for the final request. Delays could also be done server-side, to avoid blocking a bunch of threads.

Reply:
I didn't notice this was using remote. Yes, this should use loopback. In addition to being more reliable, the client and server can synchronize to achieve the exact interleavings necessary.

Reply:
Initially, I tried implementing this test using our loopback server, but it doesn't support ALPN which make WinHTTP to downgrade protocol version.  Or, do you know how to fix this?